test/uk/gov/hmrc/excisemovementcontrolsystemapi/fixture/FakeJsonParsers.scala

```diff
@@ -38,14 +39,32 @@ trait FakeJsonParsers {
         )
       )
 
+    override def refineETDS[A](
+      request: EnrolmentRequest[A]
+    ): Future[Either[Result, ParsedPreValidateTraderETDSRequest[A]]] = Future.successful(
+      Right(
+        preValidateTrader.request.ParsedPreValidateTraderETDSRequest(
+          EnrolmentRequest(request, Set("ern123"), "123"),
+          getPreValidateTraderETDSRequest
+        )
+      )
+    )
+
     override protected def executionContext: ExecutionContext = ExecutionContext.Implicits.global
+
   }
 
   object FakeFailureJsonParser extends ParseJsonAction {
     override def refine[A](request: EnrolmentRequest[A]): Future[Either[Result, ParsedPreValidateTraderRequest[A]]] =
       Future.successful(Left(BadRequest("error")))
 
+    override def refineETDS[A](
+      request: EnrolmentRequest[A]
+    ): Future[Either[Result, ParsedPreValidateTraderETDSRequest[A]]] =
+      Future.successful(Left(BadRequest("error")))
```

Comment:
It might be best with a new ticket (and I think this might an issue in general behaviour) but this doesn't match the error messaging pattern - if you look at the devhub/mdtp handbook they both have a reference to an error message standard (something like message: String, code: String)